client/components/Events/ManageEventTemplatesModal.jsx

```diff
@@ -0,0 +1,205 @@
+    deleteTemplate: PropTypes.func,
+};
+
+class ManageEventTemplatesModalComponent extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            templateInEditMode: null,
+        };
+
+        this.editTemplate = this.editTemplate.bind(this);
+        this.deleteTemplate = this.deleteTemplate.bind(this);
+        this.onEditComplete = this.onEditComplete.bind(this);
+    }
+
+    editTemplate(id) {
+        this.setState({
+            templateInEditMode: id,
+        });
+    }
+
+    onEditComplete(templateEdited) {
+        this.props.dispatch(eventsPlanning.ui.updateEventTemplate(
+            this.props.eventTemplates.find(({_id}) => _id === this.state.templateInEditMode),
+            {template_name: templateEdited.template_name}
+        ));
+
+        this.setState({
+            templateInEditMode: null,
+        });
+    }
+
+    deleteTemplate(id) {
+        this.props.modal.confirm(gettext('Confirm delete')).then(() => {
+            this.props.dispatch(eventsPlanning.ui.removeEventTemplate(
+                this.props.eventTemplates.find(({_id}) => _id === id)
+            ));
+        });
+    }
+
+    render() {
+        const {handleHide} = this.props;
+
+        return (
+            <Modal xLarge={false} show={true} onHide={handleHide}>
+                <Modal.Header>
+                    <a className="close" onClick={handleHide}>
+                        <i className="icon-close-small" />
+                    </a>
+                    <h3 className="modal__heading">{gettext('Manage Event Templates')}</h3>
+                </Modal.Header>
+                <Modal.Body noPadding={true}>
+                    {
+                        this.state.templateInEditMode == null
+                            ? (
+                                <EventTemplatesList
+                                    eventTemplates={this.props.eventTemplates}
+                                    editTemplate={this.editTemplate}
+                                    deleteTemplate={this.deleteTemplate}
+                                />
+                            )
+                            : (
+                                <EventTemplateEdit
+                                    template={this.props.eventTemplates.find(
+                                        ({_id}) => _id === this.state.templateInEditMode)
+                                    }
+                                    onEditComplete={this.onEditComplete}
+                                />
+                            )
+                    }
+                </Modal.Body>
+            </Modal>
+        );
+    }
+}
+
+ManageEventTemplatesModalComponent.propTypes = {
+    handleHide: PropTypes.func,
+    eventTemplates: PropTypes.array,
+    modal: PropTypes.object,
+    dispatch: PropTypes.func,
+};
+
+function mapStateToProps(state) {
+    return {
+        eventTemplates: state.events.eventTemplates,
```

Comment:
Please create a selector in `client/selectors/events.js` instead of directly accessing state.